sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -188,22 +150,195 @@ export class EventHubConsumerClient {
+      );
+
+      eventProcessor = new EventProcessor(
+        consumerGroupName1,
+        this._eventHubClient,
+        partitionProcessorType,
+        optionsOrPartitionIdsOrPartitionManager3 as PartitionManager,
+        possibleOptions4
+      );
+    } else {
+      // #1: subscribe overload - read from all partitions, don't coordinate
+      log.consumerClient("Subscribing to all partitions, don't coordinate.");
+
+      const partitionProcessorType = createPartitionProcessorType(
+        onReceivedEvents2,
+        optionsOrPartitionIdsOrPartitionManager3 as SubscriptionOptions
+      );
+
+      eventProcessor = new EventProcessor(
+        consumerGroupName1,
+        this._eventHubClient,
+        partitionProcessorType,
+        new InMemoryPartitionManager(),
+        {
+          ...optionsOrPartitionIdsOrPartitionManager3 as SubscriptionOptions,
+          partitionLoadBalancer: new GreedyPartitionLoadBalancer()
+        }
+      );
+    }
+
+    eventProcessor.start();
+
+    return {
+      isReceiverOpen: () => eventProcessor.isRunning(),
+      stop: () => eventProcessor.stop(),
+      consumerGroup: () => consumerGroupName1
+    };
   }
 }
+
+/**
+ * @ignore
+ */
+export function createPartitionProcessorType(
+  onReceivedEvents: (
+    receivedEvents: ReceivedEventData[],
+    context: PartitionContext
+  ) => Promise<void>,
+  options: SubscriptionOptions = {}
+): typeof PartitionProcessor {
+  class DefaultPartitionProcessor extends PartitionProcessor {
+    async processEvents(events: ReceivedEventData[]): Promise<void> {
+      await onReceivedEvents(events, {
+        partitionId: this.partitionId,
+        consumerGroupName: this.consumerGroupName,
+        eventHubName: this.eventHubName
+      });
+    }
+
+    async processError(error: Error): Promise<void> {
+      if (options.onError) {
+        await options.onError(error, {
+          partitionId: this.partitionId,
+          consumerGroupName: this.consumerGroupName,
+          eventHubName: this.eventHubName
+        });
+      }
+    }
+
+    async initialize() {
+      if (options.onInitialize) {
+        await options.onInitialize({
+          partitionId: this.partitionId,
+          consumerGroupName: this.consumerGroupName,
+          eventHubName: this.eventHubName
+        });
+      }
+    }
+
+    async close(reason: CloseReason) {
+      if (options.onClose) {
+        await options.onClose(reason, {
+          partitionId: this.partitionId,
+          consumerGroupName: this.consumerGroupName,
+          eventHubName: this.eventHubName
+        });
+      }
+    }
+  }
+
+  return DefaultPartitionProcessor;
+}
+
+/**
+ * @internal
+ * @ignore
+ */
+export function isPartitionManager(possible: SubscriptionOptions | undefined | string[] | PartitionManager): possible is PartitionManager {
+
+  if (!possible) {
+    return false;
+  }
+
+  const partitionManager = possible as PartitionManager;
+
+  return partitionManager.claimOwnership != null
```

Comment:
A safer check might be to test if these are functions (since that's what we really need) using typeof X === 'function'

Reply:
Agreed. Added.